core/server/src/main/java/alluxio/worker/netty/DataServerReadHandler.java

```diff
@@ -137,6 +141,11 @@ public void channelRead(ChannelHandlerContext ctx, Object object) throws Excepti
         mLock.lock();
         try {
           mRequest.mCancelled = mPosToQueue;
+          if (remainingToWrite() <= 0) {
```

Comment:
No. it should not be possible because mCancelled and mPosToQueue are both updated with mLock acquired.